### PR TITLE
Do not play simultaneous walk/fly sounds for battle speed 9 and 10

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR is a part of #7322 and changes only the move sound playback logic for battle speeds 9 and 10.
The other part of that PR needs some more work, especially for 6-8 battle speed.

On these battle speeds most creatures should finish their movement until the move sound is ended. Playing multiple simultaneous sounds makes the walk/fly sounds like a lot of loud echos.
For the case when creature does not finish its movement and the walk/fly sound has ended (on low performance systems with low frame rate) the next sound will be played after the previous one is finished.

### CODE REVIEW

**Comment by Districh-ru** [comment]
Hello @ihhub, @Branikolog, @oleg-derevenetz, @zenseii and @LeHerosInconnu!

I would like to hear your opinion on this implementation of limiting simultaneous walk/fly sounds for creatures on battle speeds 9 and 10. Is it acceptable or maybe you have some ideas how to make it better.

I will be on vacation in nature for about two weeks away from normal internet (only poor 2G mobile connection), so I won't be able to reply to your comments soon. :( You can take your time to check to the proposed approach. :)

Also, if you have time, could you please comment the approach used for 5-8 battle speeds in #7322.

Thank you in advance and good luck with the version 1.0.7 release!

**Comment by oleg-derevenetz** [comment]
> I will be on vacation in nature for about two weeks

Happy vacation!

**Comment by Districh-ru** [comment]
> Hi, @Districh-ru. What is the status of this PR? I'm asking because I saw in the other PR that an issue with SDL-Mixer got addressed, but does that affect this PR and the approach chosen in it? If not, then I will go ahead and review it.

Hi, @zenseii. This PR is ready for review. Please, go ahead with it when you have time. :)
It has changes only for very-very fast battle speeds (9 and 10) where only one sound proposed to be played during unit movement. As playing a new sound for every new cell move produces only a mess.
This PR does not use SDL_mixer sound fade functions because automatically fading the only sound sometimes sounds not good. In future we should check every move sound and find timestamps fore every step in the file to properly stop or fade the sound (e.g. not in the middle of step).

Thanks to @oleg-derevenetz, SDL2_mixer brunch now has a fix for a case when fade was called just at the end of sound. But yet SDL_mixer do not do accurate fade (https://github.com/ihhub/fheroes2/pull/7322#issuecomment-1601106355). So for now IMHO we should not use it and possibly implement fade in our engine and use it after we get all proper timestamps for all moving sounds.
I'll start to analyze the sound files in my free time a little later.

**Comment by oleg-derevenetz** [comment]
> SDL2_mixer brunch now has a fix for a case when fade was called just at the end of sound

I should note that, although it has this fix in its `master` and `SDL2` branches in repo, there is no new release with this fix yet.

**Comment by ihhub** [comment]
Hi @Districh-ru and @oleg-derevenetz , are we going to wait for SDL to make a new release in order to merge this pull request?

**Comment by Districh-ru** [comment]
Hi, @ihhub, this PR does not need the mentioned SDL_mixer fix.
It can be merged with the current SDL_mixer release.

**Comment by ihhub** [comment]
Hi @zenseii , could you please review this pull request today if you have time so we can merge it for the upcoming release?

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -3964,11 +3966,17 @@ void Battle::Interface::RedrawActionMove( Unit & unit, const Indexes & path )
         }
     }
 
+    // For Battle speed 9 and 10 we play one sound at time without any simultaneous playbacks
+    // as on these speeds the unit will end its movement before the sound is finished.
```
[the remark this answers is not in the document]
Comment (Districh-ru) [reply src/fheroes2/battle/battle_interface.cpp]: Hi, @zenseii, thanks for correcting my grammar!

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thanks a lot for this improvement!

**Pull request merged**